meson.build

```diff
@@ -0,0 +1,176 @@
+####################################################################################################################################
+# pgBackRest Project
+####################################################################################################################################
+project(
+    'pgbackrest',
+    ['c'],
+    version: '2.40dev',
+    license: 'MIT',
+    meson_version: '>=0.53.2',
+    default_options: [
+        # Core options
+        'buildtype=release',
+        'warning_level=2',
+
+        # Base options
+        'b_ndebug=if-release',
+
+        # Compiler options
+        'c_std=c99',
+    ],
+)
+
+####################################################################################################################################
+# OS-specific settings
+####################################################################################################################################
+if host_machine.system() == 'linux'
+    add_global_arguments('-D_POSIX_C_SOURCE=200809L', language : 'c')
+elif host_machine.system() == 'darwin'
+    add_global_arguments('-D_DARWIN_C_SOURCE', language : 'c')
+endif
+
+####################################################################################################################################
+# Enable as many additional warnings as possible to catch potential errors
+####################################################################################################################################
+warning_enable = [
+    # Warn for implicit conversions that may alter a value
+    'conversion',
+
+    # Warn about duplicated conditions in an if-else-if chain
+    'duplicated-cond',
+
+    # Warn when an if-else has identical branches
+    'duplicated-branches',
+
+    # Warn if the format string is not a string literal and cannot be checked
+    'format-nonliteral',
+
+    # Enable -Wformat plus additional format checks
+    'format=2',
+
+    # Warn if the format string requires an unsigned argument and the argument is signed and vice versa
+    'format-signedness',
+
+    # Warn about anything that depends on the “size of” a function type or of void
+    'pointer-arith',
+
+    # Warn if a function is declared or defined without specifying the argument types
+    'strict-prototypes',
+
+    # Warn if a variable-length array is used
+    'vla',
+
+    # Give string constants the type const char[length] so that copying the address of one into a non-const char * pointer produces
+    # a warning
+    'write-strings',
+]
+
+foreach warning : warning_enable
+    if meson.get_compiler('c').has_argument('-W' + warning)
+        add_project_arguments('-W' + warning, language: 'c')
```

Comment:
Meson also has a `get_supported_arguments(list_of_warning_flags)` that allows you to do this:

```
add_project_arguments(cc.get_supported_arguments(warning_enable), language: 'c')
```
However, using it assumes that warning_enable is a list of flags which already have their leading `-W`. It's safe to do that for the enable flags, since you don't need to modify it to the `-Wno-*` form.

It's slightly inconsistent with the warning_disable handling below, but that may be okay since ultimately you want to use Meson 0.51.0 support for handling that automatically too :) and many people excuse inconsistent style if it's only a temporary backwards-compat measure.

This would also mean that you can later on extend warning_enable with MSVC style flags and it doesn't get the GCC-style leading `-W` treatment, so it simplifies adding multi-platform support.